fabric-registry-sync-v0/src/testmod/java/net/fabricmc/fabric/test/registry/sync/TestDynamicObject.java

```diff
@@ -0,0 +1,10 @@
+package net.fabricmc.fabric.test.registry.sync;
+
+import com.mojang.serialization.Codec;
+import com.mojang.serialization.codecs.RecordCodecBuilder;
+
+public record TestDynamicObject(String name) {
+	public static final Codec<TestDynamicObject> CODEC = RecordCodecBuilder.create(instance -> instance.group(
+			Codec.STRING.fieldOf("name").forGetter(TestDynamicObject::name)
+	).apply(instance, TestDynamicObject::new));
```

Comment:
I think it looks easier
```java
public static final Codec<TestDynamicObject> CODEC = Codec.STRING
        .fieldOf("name")
        .xmap(TestDynamicObject::new, p -> p.name)
        .codec();
```

Reply:
Test code should be something modders can copy and adapt. Since one-field data is rare I think the current code is better for that purpose.

Reply:
I don't think it really matters which variant we use here. Using `RecordCodecBuilder` might be more recognisable to people reading this code since it's used way more often in vanilla and mods than converting a single codec `fieldOf` to a full codec.